### ISSUE TITLE
Impossible to suppress language in editUrl for language-enabled sites?

### ISSUE THREAD

**Issue by JanRomero** [processwire/processwire-issues#1106]
Since updating from 2.7.x I’ve noticed the language parameter appearing at the end of my query string.

The comment on Page::editUrl() states that the `$options['language']` argument is optional, either specifies a concrete language or forces the current user’s language if `true`.

The code itself, however, forces the current user’s language by default, only overriding it if a concrete language is specified or removes it for admin pages.

`$options['language'] === false` does nothing. Neither does not passing the argument, because both will go into the `if empty()` codepath.

I believe at least passing `false` should drop the language from the query string. My personal dream would be for that to be the default as well, which is how I understand the phpdoc anyway.

I’m submitting this because I just like my urls clean and short, especially because I sometimes have my system send edit-links to users, and because it would help me see the page ID better at a glance.

Nothing should break because the admin already honors the user’s language setting.

-----

### TL;DR:
My proposal would be to [change this line](https://github.com/processwire/processwire/blob/51629cdd5f381d3881133baf83e1bd2d9306f867/wire/core/Page.php#L3320) to just `$language = null;`.

-----

Sorry I posted this in the ProcessWire repo earlier, GitHub’s “reference in new issue” feature sent me there… I consider this an issue as opposed to a request because it goes against the documentation as I understand it.

**Comment by matjazpotocnik** [comment]
@ryancramerdesign, this is still present in 3.0.271. Passing ['language' => false] has no effect because the value is checked with empty(). Outside the admin, editUrl() still falls back to the current user’s language and adds it to the URL. At minimum, an explicit false should suppress the language parameter.

Possible fix:

```diff
diff --git a/wire/core/Page/PageTraversal.php b/wire/core/Page/PageTraversal.php
--- a/wire/core/Page/PageTraversal.php
+++ b/wire/core/Page/PageTraversal.php
@@
- *  - `language` (Language|bool): Optionally specify Language to start editor in, or boolean true to force current user language.
+ *  - `language` (Language|bool): Optionally specify Language to start editor in, boolean true to force current user language,
+ *    or boolean false to omit the language from the URL.
@@
         if($languages) {
             $language = $page->wire()->user->language;
-			if(empty($optionsArray['language'])) {
+			if(isset($optionsArray['language']) && $optionsArray['language'] === false) {
+				$language = null;
+			} else if(empty($optionsArray['language'])) {
                 if($page->wire()->page->template->id == $adminTemplate->id) $language = null;
             } else if($optionsArray['language'] instanceof Page) {
```

```diff
diff --git a/wire/core/Page/Page.php b/wire/core/Page/Page.php
--- a/wire/core/Page/Page.php
+++ b/wire/core/Page/Page.php
@@
-	 *  - `language` (Language|bool): Optionally specify Language to start editor in, or boolean true to force current user language.
+	 *  - `language` (Language|bool): Optionally specify Language to start editor in, boolean true to force current user language,
+	 *    or boolean false to omit the language from the URL.
```

**Comment by ryancramerdesign** [comment]
Confirmed and fixed as suggested. Verified: `$page->editUrl(['language' => false])` now omits the `&language=` parameter entirely, while `$page->editUrl()` (no option) is unchanged.

Fixed in 18f01e1d.

-Claude
